src/java.base/share/classes/java/util/zip/Adler32.java

```diff
@@ -96,7 +99,7 @@ public void update(ByteBuffer buffer) {
         if (rem <= 0)
             return;
         if (buffer.isDirect()) {
-            try {
+            try (var sessionAcquisition = NIO_ACCESS.acquireSessionAsAutoCloseable(buffer)) {
```

Comment:
Eventually, I hope most of them will be named `_`.

Reply:
maybe just `bufferLock` and and just `acquireBuffer` ?

Reply:
Would it be possible to re-examine acquireSession returning Runnable and 
acquireSessionAsAutoCloseable returning AutoCloseable. The naming is awkward at most of the use sites and maybe we can do better.

Reply:
I think it would be confusing to have overloads with the same name. 

The `aquireSession(Buffer, boolean)` method is only used once and is used in `IOUtil` so we could rename it to `aquireSessionOrNull` and then rename `acquireSessionAsAutoCloseable` to `acquireSession`.

Reply:
IIRC, Runnable was chosen back then because of the issues with using TWR which was generating warnings (because resource not accessed in body of TWR). I don't have strong opinions on this, other than we should only have one way to do things, rather than 2-3. Since the JDK has capabilities to acquire and release a session w/o a TWR and/or Runnable, one might also consider whether exposing acquire/release separately, and just use those.